File: 프로그래머스/2/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.py
import math
def solution(r1, r2):
    result=0
    for i in range(r2+1):
        l=math.floor(math.sqrt(r2**2-i**2))
        r=math.ceil(math.sqrt(r1**2-i**2))if i<=r1 else 0
        result+=l-r+1
    return result*4-4*(r2-r1+1)

File: 프로그래머스/2/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <math.h>

long long solution(int r1, int r2) {
    long long r,l,result=0;
    for(int i=0; i<r2+1; i++){
        l=floor(sqrt(pow(r2,2)-pow(i,2)));
        r=(i<=r1)?ceil(sqrt(pow(r1,2)-pow(i,2))):0;
        result+=l-r+1;
    }
    return result*4-4*(r2-r1+1);
}
